astrocut/cube_cut.py

```diff
@@ -845,6 +847,9 @@ def cube_cut(
         if verbose:
             start_time = time()
 
+        # Declare the product type being used to make the cutouts
+        self.product = product
```

Comment:
why is `product` an argument of cube_cut() function still? 

self.product should only be assigned during initialization of the cutout factory, now, right?

Reply:
`self.product` is assigned to `None` when the `CutoutFactory` class is initialized. It's then replaced by "TICA" or "SPOC" depending on what value the user passes when calling `cube_cut`. But this reminds me, I should enforce that the "TICA" or "SPOC" string is in all uppercase so we don't run into a case-sensitive bug down the line.

Reply:
i see, i missed that originally -- i guess i'd expected it to be an argument to the initialization of the coutoutfactory instance.

in general, i think this changeset is moving pretty far away from the idea of a factory class and we're now adding a bunch of conditionals if / else throughout the codebase which i'm not a huge fan of. 

perhaps we should hold off on refactoring the product and first just implement the change to remove the error array from TICA cubes?

When refactoring, i think there might be some ways to try and avoid all the conditional logic that is in this PR, which gets very hard to maintain.

Reply:
We've already removed the error array from the TICA cubes in PR #87 

I ran some tests and confirmed that the removal of the error arrays from the TICA cubes is worth the hassle of refactoring things in `CutoutFactory` (cubes are now 1/2 the size and take about 1/3 less time to make). I thought about making a new class just for TICA called `TicaCutoutFactory`, the same way things exist separately in `make_cube.py`, but it leads to a lot of redundancies.

Reply:
sorry, i mean cutouts, not cubes, above.

i'm not disagreeing about removing the error component from the cutout (or the cube). 

i wonder, do fits files support sparse arrays?

Reply:
You can make sparse matrices as numpy arrays, so I assume if `fits` can support a numpy array, they can support a sparse array if numpy is its data type.

Reply:
I count 11 conditionals where we're checking if the cutout product is TICA or SPOC when making a cutout and I'm worried about the added complexity that has been added to this codebase by adding just a single new product type.

What I would like is a lot less conditional behavior based on the product and to instead utilize the factory based structure this module purports to be to avoid all these conditionals. 

We can handle this in a separate PR / ticket if you like. I do think this is important for maintainability, both for the cube creation as well as the cutouts.  I did mention some of this when we added TICA to the cube creation module as well.

With some refactoring, we can possibly simplify this to use the factory methodology and define objects (SPOC / TICA) that use the factory to make the cutouts, and can keep the different behavior isolated to their own class methods.  If you like, we can work together to sketch this out.

I think we should not undertake a large refactoring in this PR, so I will instead focus my comments on the specifics of not returning the error array when making TICA cutouts.

Reply:
I see what you mean, and yes it makes sense to do this refactoring in a separate PR. This PR has a smaller time-constraint, as we have to start generating the errorless TICA cubes as soon as Aug. 14th, so I'd like to have this merged by the end of this week (Aug. 3rd) ideally. That way @hmlewis-astro can start generating the error-free TICA cubes asynchronously while I work on simplifying the code. Let me know if you agree with this and I'll make a new Jira ticket for the refactoring @falkben

Reply:
That sounds good. 

Also, this might be related: https://github.com/spacetelescope/astrocut/issues/62 since there was a similar discussion when the make cube stuff was added for TICA https://github.com/spacetelescope/astrocut/pull/59#discussion_r890526958

I'm not sure what makes sense re: refactoring, whether class inheritance would actually make things more maintainable etc, but we might want to have these work items linked, somehow.  

I'd be happy to sketch this out with you at some point to try and generalize this a bit.

Reply:
OK, I will link all three PRs (#88, #62, and #59) in the Astrocut Refactoring Jira ticket. Thank you!